tests/test_core.py

```diff
@@ -92,33 +93,49 @@ def __init__(self, dt: int, timing: dict) -> None:
         self.action_space = ngym.spaces.Discrete(2)
 
     def _new_trial(self, **kwargs):
+        # Add periods
+        if self.timing:
+            self.add_period(list(self.timing.keys()))
+            self.add_ob(1)
+
         return {}
 
     def _step(self, action):  # noqa: ARG002
         return self.ob_now, 0, False, False, {"new_trial": False}
 
 
 @pytest.mark.parametrize(
-    ("timing", "expected_timesteps"),
+    ("timing", "expected_stats"),
     [
-        # Simple integer/float timing values
-        ({"fixation": 300, "stimulus": 500, "decision": 200}, 10),
-        # Uniform distribution
-        ({"fixation": 300, "stimulus": ["uniform", [400, 600]], "decision": 200}, 10),
-        # Truncated exponential distribution
-        ({"fixation": 300, "stimulus": ["truncated_exponential", [400, 100, 300, 500]], "decision": 200}, 9),
-        # Choice distribution
-        ({"fixation": 300, "stimulus": ["choice", [300, 400, 500]], "decision": 200}, 9),
-        # List of values for random choice
-        ({"fixation": 300, "stimulus": [400, 500, 600], "decision": 200}, 10),
-        # Constant value
-        ({"fixation": 300, "stimulus": ["constant", 450], "decision": 200}, int(9.5)),
-        # Empty timing
-        ({}, 0),
+        # Fixed timing (exact values)
+        (
+            {"fixation": 300, "stimulus": 500, "decision": 200},
+            {"mean": 10, "std": 0, "percentile_95": 10, "max": 10},
+        ),
+        (
+            {},
+            {"mean": 0, "std": 0, "percentile_95": 0, "max": 0},
+        ),
+        # Randomized timing (specific expected values, after fixing the random seed)
+        (
+            {"fixation": 300, "stimulus": ["uniform", [400, 600]], "decision": 200},
+            {"mean": 10, "std": 0, "percentile_95": 10, "max": 10},
+        ),
+        (
+            {"fixation": 300, "stimulus": ["truncated_exponential", [400, 100, 500]], "decision": 200},
+            {"mean": 8, "std": 1, "percentile_95": 8, "max": 8},
+        ),
+        (
+            {"fixation": 300, "stimulus": ["choice", [300, 400, 500]], "decision": 200},
+            {"mean": 9, "std": 0, "percentile_95": 9, "max": 9},
+        ),
     ],
 )
-def test_avg_timesteps_per_trial(timing, expected_timesteps):
-    """Test the avg_timesteps_per_trial property with different timing configurations."""
+def test_trial_length_stats(timing, expected_stats):
+    """Test trial length stats for both fixed and randomized timing configurations."""
     dt = 100
-    env = TestEnv(dt=dt, timing=timing)
-    assert env.avg_timesteps_per_trial == expected_timesteps
+    env = TestEnv(dt=dt, timing=timing, seed=123)  # <-- Fix the seed here
```

Comment:
what's the point of fixing the seed here, if a default seed is also specified above